### PR TITLE
split registration permissions

### PR DESCRIPTION
**Pull request by haoqing0110**
Related PR: https://github.com/open-cluster-management-io/registration-operator/pull/250
Signed-off-by: haoqing0110 <qhao@redhat.com>

### CODE REVIEW

File: deploy/spoke/clusterrole_addition.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  name: open-cluster-management:spoke-addition
+rules:
+# Allow agent to get/list/watch/create/delete/update/patch secrets.
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "list", "watch", "create", "delete", "update", "patch"]
+# Allow agent to list addons lease
+- apiGroups: ["coordination.k8s.io"]
+  resources: ["leases"]
+  verbs: ["create", "get", "list", "update", "watch", "patch"]
```
Comment (skeeey) [inline deploy/spoke/clusterrole_addition.yaml]: please add an empty line in the end of file

File: deploy/spoke/clusterrole.yaml [hunk 2]
```diff
@@ -3,20 +3,12 @@ kind: ClusterRole
...
-  verbs: ["create"]
 # Allow agent to list clusterclaims
 - apiGroups: ["cluster.open-cluster-management.io"]
   resources: ["clusterclaims"]
-  verbs: ["get", "list", "watch"]
-# Allow agent to list addons lease
-- apiGroups: ["coordination.k8s.io"]
-  resources: ["leases"]
-  verbs: ["get", "list", "watch"]
+  verbs: ["get", "list", "watch"]
```
Comment (skeeey) [inline deploy/spoke/clusterrole.yaml]: please add an empty line in the end of file

File: deploy/spoke/role_extension-apiserver.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: open-cluster-management:registration-agent:extension-apiserver
+  namespace: kube-system
+rules:
+- apiGroups: [""]
+  resources: ["configmaps"]
+  resourceNames: ["extension-apiserver-authentication"]
+  verbs: ["get", "list", "watch"]
```
Comment (skeeey) [inline deploy/spoke/role_extension-apiserver.yaml]: please add an empty line in the end of file

File: deploy/spoke/clusterrole_addition.yaml (hunk 1 above)
Comment (haoqing0110) [reply deploy/spoke/clusterrole_addition.yaml]: fixed

File: deploy/spoke/clusterrole.yaml (hunk 2 above)
Comment (haoqing0110) [reply deploy/spoke/clusterrole.yaml]: fixed

File: deploy/spoke/role_extension-apiserver.yaml (hunk 3 above)
Comment (haoqing0110) [reply deploy/spoke/role_extension-apiserver.yaml]: fixed

File: deploy/spoke/kustomization.yaml [hunk 4]
```diff
@@ -30,8 +30,12 @@ resources:
 - ./service_account.yaml
 - ./clusterrole.yaml
 - ./clusterrole_binding.yaml
+- ./clusterrole_addon-management.yaml
+- ./clusterrole_binding_addon-management.yaml
 - ./role.yaml
 - ./role_binding.yaml
+#- ./role_extension-apiserver.yaml
```
Comment (qiujian16) [inline deploy/spoke/kustomization.yaml]: why comment out these line?

Comment (haoqing0110) [reply deploy/spoke/kustomization.yaml]: these line moved to Makefile, will remove them.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**